sympy/solvers/polysys.py

```diff
@@ -228,7 +241,16 @@ def _subs_root(f, gen, zero):
     def _solve_reduced_system(system, gens, entry=False):
         """Recursively solves reduced polynomial systems. """
         if len(system) == len(gens) == 1:
-            zeros = list(roots(system[0], gens[-1]).keys())
+            rts = roots(system[0], gens[-1])
+
+            if sum(rts.values()) < degree(system[0], gens[-1]) and strict:
```

Comment:
The order should be reversed here like
```
if strict and ...
```
to avoid computing degree etc if strict is False.